test/cases/api.sh

```diff
@@ -953,4 +953,30 @@ curl \
         --header "x-rh-identity: $INVALIDAUTHSTRING" \
         http://localhost:443/api/composer/v1/version)" = "404" ]
 
+#
+# Make sure that a request that makes the dnf-json crash returns a 500 error
+#
+[ "$(curl \
+    --silent \
+    --output /dev/null \
+    --write-out '%{http_code}' \
+    -k --key /etc/osbuild-composer/client-key.pem \
+    --cert /etc/osbuild-composer/client-crt.pem \
+    -H "Content-Type: application/json" \
+    -d '{"distribution":"centos-8","image_requests":[{"architecture":"x86_64","image_type":"ami"}]}' \
```

Comment:
I'm not sure about the 500 test case here. Since it's kinda weird it gets to trying depsolving in the first place when the request itself is invalid. So the ideal scenario should be 400 here (but let's improve input validation later).

How about having a valid request, but forcing dnf-json to crash? We could:
1. mv /usr/libexec/osbuild-composer/dnf-json /usr/libexec/osbuild-composer/dnf-json.bak
2. write a python3 executable to /usr/libexec/osbuild-composer/dnf-json which always crashes, just something like:
```
#!/usr/bin/python3
raise Exception
```
3. check that it returns 500 now
4. mv -f /usr/libexec/osbuild-composer/dnf-json.bak /usr/libexec/osbuild-composer/dnf-json

thoughts?

Reply:
I agree. Once we will have input validation this testcase will fail.
We can't mv the dnf-json because current user don't have access rights to the script. We need to find another way.

Reply:
Ok, since the script has sudo access I've done the thing :)